### ISSUE TITLE
Flutter example app runs on simulator but not iOS device 

### ISSUE THREAD

**Issue by lukepighetti** [flutter/flutter#18505]
## Steps to Reproduce

1. install flutter
2. install vsc and flutter extension
3. run default project, works on simulator, fails on device

Note: Dart extension is installed on VSC but it's not registering in Flutter Doctor.

## Logs

```
Lukes-iMac:quiz lukepighetti$ flutter run
More than one device connected; please specify a device with the '-d <deviceId>' flag, or use '-d all' to act on all devices.

Luke's iPhone • 0bc0c4864d8e08aac4456893cded68bf0b6767d5 • ios • iOS 11.4
iPhone X      • 97AC0229-A8BD-4704-B61F-DF517A59EEC0     • ios • iOS 11.4 (simulator)
Lukes-iMac:quiz lukepighetti$ flutter run -d all
Launching lib/main.dart on Luke's iPhone in debug mode...
Automatically signing iOS for device deployment using specified development team in Xcode project: TA8U33SGB8
Starting Xcode build...
 ├─Assembling Flutter resources...                    1.1s
 └─Compiling, linking and signing...                  4.0s
Xcode build done.                                            6.8s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **

Xcode's output:
↳
    === BUILD TARGET Runner OF PROJECT Runner WITH CONFIGURATION Debug ===
    While building module 'Flutter' imported from /Users/lukepighetti/Library/Mobile Documents/com~apple~CloudDocs/code/flutter/quiz/ios/Runner/GeneratedPluginRegistrant.h:8:
    In file included from <module-includes>:1:
    In file included from /Users/lukepighetti/Library/Mobile Documents/com~apple~CloudDocs/code/flutter/quiz/ios/Flutter/Flutter.framework/Headers/Flutter.h:37:
    In file included from /Users/lukepighetti/Library/Mobile Documents/com~apple~CloudDocs/code/flutter/quiz/ios/Flutter/Flutter.framework/Headers/FlutterAppDelegate.h:11:
    /Users/lukepighetti/Library/Mobile Documents/com~apple~CloudDocs/code/flutter/quiz/ios/Flutter/Flutter.framework/Headers/FlutterPlugin.h:140:58: warning: this block declaration is not a prototype [-Wstrict-prototypes]
                          completionHandler:(nonnull void (^)())completionHandler;
                                                             ^
                                                              void
    1 warning generated.
    1 warning generated.
    === BUILD TARGET Runner OF PROJECT Runner WITH CONFIGURATION Debug ===
    While building module 'Flutter' imported from /Users/lukepighetti/Library/Mobile Documents/com~apple~CloudDocs/code/flutter/quiz/ios/Runner/AppDelegate.h:2:
    In file included from <module-includes>:1:
    In file included from /Users/lukepighetti/Library/Mobile Documents/com~apple~CloudDocs/code/flutter/quiz/ios/Flutter/Flutter.framework/Headers/Flutter.h:37:
    In file included from /Users/lukepighetti/Library/Mobile Documents/com~apple~CloudDocs/code/flutter/quiz/ios/Flutter/Flutter.framework/Headers/FlutterAppDelegate.h:11:
    /Users/lukepighetti/Library/Mobile Documents/com~apple~CloudDocs/code/flutter/quiz/ios/Flutter/Flutter.framework/Headers/FlutterPlugin.h:140:58: warning: this block declaration is not a prototype [-Wstrict-prototypes]
                          completionHandler:(nonnull void (^)())completionHandler;
                                                             ^
                                                              void
    1 warning generated.
    1 warning generated.
    === BUILD TARGET Runner OF PROJECT Runner WITH CONFIGURATION Debug ===
    /Users/lukepighetti/Library/Mobile Documents/com~apple~CloudDocs/code/flutter/quiz/build/ios/Debug-iphoneos/Runner.app: resource fork, Finder information, or similar detritus not allowed
    Command /usr/bin/codesign failed with exit code 1
Could not build the precompiled application for the device.

Error launching application on Luke's iPhone.
```

```
Lukes-iMac:quiz lukepighetti$ flutter doctor -v
[✓] Flutter (Channel unknown, v0.0.0-unknown, on Mac OS X 10.13.5 17F77, locale en-US)
    • Flutter version 0.0.0-unknown at /Users/lukepighetti/flutter
    • Framework revision 3c0e1add7a (20 hours ago), 2018-06-14 13:36:20 -0400
    • Engine revision 06afdfe54e
    • Dart version 2.0.0-dev.54.0.flutter-46ab040e58

[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.io/setup/#android-setup for detailed instructions).
      If Android SDK has been installed to a custom location, set $ANDROID_HOME to that location.

[✓] iOS toolchain - develop for iOS devices (Xcode 9.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 9.4.1, Build version 9F2000
    • ios-deploy 1.9.2
    • CocoaPods version 1.5.3

[✗] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.io/setup/#android-setup for detailed instructions).

[!] VS Code (version 1.24.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Dart Code extension not installed; install from
      https://marketplace.visualstudio.com/items?itemName=Dart-Code.dart-code

[✓] Connected devices (2 available)
    • Luke's iPhone • 0bc0c4864d8e08aac4456893cded68bf0b6767d5 • ios • iOS 11.4
    • iPhone X      • 97AC0229-A8BD-4704-B61F-DF517A59EEC0     • ios • iOS 11.4 (simulator)

! Doctor found issues in 3 categories.
```

**Comment by lukepighetti** [comment]
I found this documentation https://developer.apple.com/library/archive/qa/qa1940/_index.html

I tried `sudo xattr -cr .` in project directory because without sudo I didn't have permission. This solved it temporarily. Problem came back almost immediately.

Then I remembered that my project directory was in an iCloud directory. So I moved the folder out and had to run `sudo xattr -cr .` again. This solved it, and the solution seems to be sticking.
